compiler/plugins/input/StableHLO/stablehlo-iree/Conversion/StableHLOCustomCalls.cpp

```diff
@@ -221,6 +310,242 @@ struct HouseholderReflectorRewriter final
   }
 };
 
+llvm::SmallVector<OpFoldResult> makeIndexArray(llvm::SmallVector<int64_t> idx,
+                                               ImplicitLocOpBuilder b) {
+  llvm::SmallVector<OpFoldResult> res;
+  res.reserve(idx.size());
+  for (auto i : idx) {
+    res.push_back(b.getIndexAttr(i));
+  }
+
+  return res;
+}
+
+Value extractSlice(Value v, llvm::ArrayRef<OpFoldResult> offsets,
+                   llvm::ArrayRef<int64_t> sizes, llvm::ArrayRef<int64_t> dims,
+                   ImplicitLocOpBuilder b) {
+  assert(offsets.size() == dims.size());
+  assert(sizes.size() == dims.size());
+
+  auto vTy = cast<ShapedType>(v.getType());
+  auto rank = vTy.getRank();
+
+  llvm::SmallVector<OpFoldResult> vOffsets(rank, b.getIndexAttr(0));
+  llvm::SmallVector<OpFoldResult> vStrides(rank, b.getIndexAttr(1));
+
+  llvm::SmallVector<int64_t> shape(vTy.getShape());
+  for (int i = 0; i < dims.size(); ++i) {
+    shape[dims[i]] = sizes[i];
+    vOffsets[dims[i]] = offsets[i];
+  }
+
+  llvm::SmallVector<OpFoldResult> vSizes;
+  for (auto v : shape) {
+    vSizes.push_back(b.getIndexAttr(v));
+  }
+
+  // Slice the top-left most element:
+  auto sliceTy = RankedTensorType::get(shape, vTy.getElementType());
+  return b.create<tensor::ExtractSliceOp>(sliceTy, v, vOffsets, vSizes,
+                                          vStrides);
+}
+
+Value insertSlice(Value dest, Value update,
+                  llvm::ArrayRef<OpFoldResult> offsets,
+                  llvm::ArrayRef<int64_t> dims, ImplicitLocOpBuilder b) {
+  assert(offsets.size() == dims.size());
+
+  auto destTy = cast<ShapedType>(dest.getType());
+  auto updateTy = cast<ShapedType>(update.getType());
+  auto rank = destTy.getRank();
+
+  llvm::SmallVector<OpFoldResult> vOffsets(rank, b.getIndexAttr(0));
+  llvm::SmallVector<OpFoldResult> vStrides(rank, b.getIndexAttr(1));
+
+  llvm::SmallVector<int64_t> shape(updateTy.getShape());
+  for (int i = 0; i < dims.size(); ++i) {
+    vOffsets[dims[i]] = offsets[i];
+  }
+
+  llvm::SmallVector<OpFoldResult> vSizes;
+  for (auto v : shape) {
+    vSizes.push_back(b.getIndexAttr(v));
+  }
+
+  return b.create<tensor::InsertSliceOp>(update, dest, vOffsets, vSizes,
+                                         vStrides);
+}
+
+AffineMap getAffineMap(llvm::ArrayRef<int64_t> skipDims, int64_t rank,
```

Comment:
This is a very generic name so comment would help (I saw you marked some static and some not, not sure if on purpose).